projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -202,16 +260,57 @@ export class CartesianLegend {
     this.activeSeriesSubject.next();
   }
 
-  private updateActiveSeries(seriesEntry: Series<{}>): void {
-    if (this.isDefault) {
-      this.activeSeries = [];
-      this.activeSeries.push(seriesEntry);
-      this.isDefault = false;
+  private getGroupedSeries(): Series<{}>[][] {
+    if (!this.isGrouped) {
+      return [];
+    }
+
+    const nonHiddenSeries = this.series.filter(series => !series.hide);
+    const groupedSeries: Series<{}>[][] = [];
+    const currentGroup: Series<{}>[] = [];
+
+    nonHiddenSeries.forEach(seriesEntry => {
+      if (
+        currentGroup.length === 0 ||
+        currentGroup[currentGroup.length - 1].assignedGroup!.id === seriesEntry.assignedGroup!.id
+      ) {
+        currentGroup.push(seriesEntry);
+      } else {
+        groupedSeries.push([...currentGroup]);
```

Comment:
Are we assuming an order? Looks like it, easy to avoid.

Reply:
Used` Dictionary<Series<TData>[]>`

Reply:
We need to assume the order, otherwise I am not sure, how are we going to do it?